example/spec/loadTest/ingestLoad_queueGranPassThruSF.js

```diff
@@ -0,0 +1,238 @@
+      let testSuffix;
+      let collection;
+      let provider;
+      let inputPayload;
+
+      beforeAll(async () => {
+        try {
+          lambdaStep = new LambdaStep();
+
+          const testId = createTimestampedTestId(config.stackName, 'GranuleIngestLoadTest');
+          testSuffix = createTestSuffix(testId);
+
+          collection = { name: `MOD09GQ${testSuffix}`, version: '006' };
+          provider = { id: `s3_provider${testSuffix}` };
+
+          // populate collections, providers and test data
+          await Promise.all([
+            addCollections(
+              config.stackName,
+              config.bucket,
+              collectionsDir,
+              testSuffix
+            ),
+            addProviders(
+              config.stackName,
+              config.bucket,
+              providersDir,
+              config.bucket,
+              testSuffix
+            ),
+          ]);
+          await updateCollection({
+            prefix: config.stackName,
+            collection,
+            updateParams: { duplicateHandling: 'replace' },
+          });
+
+          const inputPayloadJson = fs.readFileSync(inputPayloadFilename, 'utf8');
+          // update test data filepaths
+          inputPayload = await setupLoadTestGranuleForIngestPassthrough(
+            config.bucket,
+            inputPayloadJson,
+            granuleCountPerWorkflow,
+            granuleRegex,
+            testSuffix
+          );
+          testSuffixes.push(testSuffix);
+
+          workflowExecution = await buildAndExecuteWorkflow(
+            config.stackName,
+            config.bucket,
+            workflowName,
+            collection,
+            provider,
+            inputPayload
+          );
+
+          testQueueGranulesExecutionArns.push(workflowExecution.executionArn);
+          testGranulesPerInputPayloads.push(inputPayload);
+        } catch (error) {
+          beforeAllFailed = `===== Block beforeAll() failed =====\n ${error}`;
+          throw new Error(beforeAllFailed);
+        }
+      });
+
+      it('completes execution with success status', () => {
+        // console.log(`==== workflowExecution =====\n ${JSON.stringify(workflowExecution)}\n`);
+        expect(workflowExecution.status).toEqual('completed');
+      });
+
+      describe('the QueueGranules Lambda function', () => {
+        it('sets granule status to completed', async () => {
+          const expectedValues = ['completed', 'running'];
+          await Promise.all(
+            inputPayload.granules.map(async (granule) => {
+              const record = await waitForApiStatus(
+                getGranule,
+                {
+                  prefix: config.stackName,
+                  granuleId: granule.granuleId,
+                },
+                expectedValues
+              );
+              expect(expectedValues).toContain(record.status);
+            })
+          );
+        });
+
+        it('has expected arns output', async () => {
```

Comment:
We talked yesterday about these tests intentionally testing workflow components as opposed to "framework" components like our queueing systems, database writes, etc. (There's still discussion to be had there and adding comments and details about what these tests are doing would be very helpful to track your thought process.)

I think, even if we move forward with these tests as "workflow component tests" we should consider the assertions we're making. These seem to be saying "QueueGranules works". That's cool but we already have tests for QueueGranules. I would think tests for workflow components would explicitly validate things related to scale. e.g. "this workflow does not throw limit exceeded errors for 10,000 granules" or something. I'm not sure which assertions would be useful but that's also a part of this effort.

Reply:
Assertion logic has been updated in the latest commit.